### PR TITLE
Add functions to manipulate path extensions

### LINKED ISSUE #26: functions to handle extensions
`path_ext()`, `path_ext<-()`, `path_ext_set()`, `path_ext_remove()`

### PR DESCRIPTION
**Pull request by jimhester**
Fixes #26

### CODE REVIEW

**Comment by jimhester** [comment]
@hadley I think this is the API we discussed on slack, let me know what you think.

**Comment by jimhester** [comment]
Maybe we should also have `path_ext_add()`?

```r
path_ext_add("foo", "tar", "gz")
#> "foo.tar.gz"
```

**Comment by hadley** [comment]
I think the big question is what's the extension of `foo.tar.gz`. Is it `.gz` or `.tar.gz`?

**Comment by jimhester** [comment]
From my experience most tools split on only the last extension. e.g.

bash
```bash
> filename=file.tar.gz
> echo ${filename##*.}
gz
```

Ruby
```ruby
file = "file.tar.gz"
File.extname(file)
 => ".gz"
```

Python
```python
>>> import os
>>> os.path.splitext("file.tar.gz")
('file.tar', '.gz')
```

Javascript (nodejs)
```javascript
> path.extname("foo.tar.gz")
'.gz'
```

**Comment by hadley** [comment]
Ok, we should follow that convention then (and mention it in the docs)

File: R/path.R [hunk 1]
```diff
@@ -98,3 +97,44 @@ path_home <- function(...) {
...
+#' path_ext("file.zip")
+#' path_ext("file.tar.gz")
+#'
+#' path_ext_remove("file.tar.gz")
+#'
+#' # Only one level of extensions is removed
+#' path_ext_set(path_ext_remove("file.tar.gz"), "zip")
+path_ext <- function(path) {
+  res <- captures(path, regexpr("(?<!^|[.])[.]([^.]+)$", path, perl = TRUE))[[1]]
+  res[is.na(res)] <- ""
```
Comment (hadley) [inline R/path.R]: This made me wonder what happens if `path_tidy()` gets an `NA`. Should we warn? Error?

Comment (jimhester) [reply R/path.R]: I think it should just pass them along as NA.

**Comment by jimhester** [comment]
Thanks for reviewing!

File: R/path.R (hunk 1 above)
Comment (hadley) [reply R/path.R]: In that case, won't this happen:

```R
path_ext(NA)
#> [1] ""
```

Comment (jimhester) [reply R/path.R]: Yes, thank you for mentioning this. Now all the `path_ext()` functions are robust to NA inputs (with tests).

**Pull request merged**